src/main/index.ts

```diff
@@ -5,12 +5,17 @@ import { Preferences } from '../types';
 import { scanImagesFactory } from './scanImages';
 import { compressImagesFactory } from './compressImages';
 import { createStore, createRuntimeStore}  from './createStore';
+import { purgeDeletedSiteData } from './purgeDeletedSiteData';
 
 const serviceContainer = LocalMain.getServiceContainer().cradle;
 
-const existingImageData = serviceContainer.userData.get(COMPRESSED_IMAGE_DATA_FILE_NAME, {});
+const existingSiteData: SiteDataBySiteID = serviceContainer.userData.get(COMPRESSED_IMAGE_DATA_FILE_NAME, {});
 
-const imageDataStore = createStore(existingImageData);
+const imageDataStore = createStore(
+	purgeDeletedSiteData(existingSiteData, serviceContainer)
+);
```

Comment:
IIRC this will run every time Local starts up, correct? Which means deleted data will only be purged on restart of Local?

If so, is there any way to hook into the site deletion event from Local proper or nah? I realize that might be over-engineering a bit too much.

Reply:
I don't think Local exposes a public way to do this atm, but I agree, that would be really nice.

Reply:
There is currently a PR for Local that exposes a named action hook - `siteDeleted` - that we can make use of here. I added [a commit](feat: add "siteDeleted" action hook to remove site image data) to implement that.

One thing worth noting is that adding an action hook here doesn't mean that this will need to have a new minimum Local compatible version. If Local doesn't ever call/do the hook, it will simply get ignored. This shouldn't be a big problem since this also implements the `SiteImageData` cleanup at add-on load time.

Reply:
Heck yah, looks great!